state/applicationoffers.go

```diff
@@ -158,32 +149,35 @@ func (s *applicationDirectory) AddOffer(offer crossmodel.ApplicationOffer) (err
 		return ops, nil
 	}
 	err = s.st.run(buildTxn)
-	return errors.Trace(err)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return s.makeApplicationOffer(doc)
 }
 
 // UpdateOffer replaces an existing offer at the same URL.
-func (s *applicationDirectory) UpdateOffer(offer crossmodel.ApplicationOffer) (err error) {
-	defer errors.DeferredAnnotatef(&err, "cannot update application offer %q", offer.ApplicationName)
+func (s *applicationOffers) UpdateOffer(offerArgs crossmodel.AddApplicationOfferArgs) (_ *crossmodel.ApplicationOffer, err error) {
```

Comment:
Ooh, I haven't seen a named return with the name _. I'll use that next time I need to do funny error handling stuff.